File: python/bubble_sort.py
import copy
import sort_tools

def bubble_sort(data, low, high):
    for i in range(low, high):
        for j in range(low, high-low-1):
        	#swap
            if data[j] > data[j+1]:
                temp = data[j]
                data[j] =  data[j+1]
                data[j+1] = temp
                
@sort_tools.timeit('bubble')            
def sort(data):
    bubble_sort(data, 0, len(data))

def main():
    data = sort_tools.build_data_set(5)

    unsorted = copy.deepcopy(data)
    sort(unsorted)

if __name__ == "__main__":
    main()

File: javascript/bubble-sort.js
function bubbleSort(array, low, high){
    for (let i = low; i < high; i++)
        for (let j = low; j < high-i-1; j++)
        {
            if ( array[j] > array[j+1] )
            {
                let temp = array[j];
                array[j] = array[j+1];
                array[j+1] = temp;
            }
        }
}

function sort(array){
    bubbleSort(array, 0, array.length);
}